### PR TITLE
Clean up warnings

### PR DESCRIPTION
**Pull request by igormp**
Continuing work on #6189.

### CODE REVIEW

**Comment by hcho3** [comment]
Can you address the issues found by the style checker? See the log at https://github.com/dmlc/xgboost/pull/6325/checks?check_run_id=1330040466.

**Comment by igormp** [comment]
Done.
Just out of curiosity, I compared the amount of warnings since the beginning of this month until now, and we went from 2340 warnings to 1889.
If we ignore the ones related to dmlc, then we went from 1584 to 1057, we managed to reduce a third of the warnings, which is pretty nice :partying_face:

File: src/common/version.cc [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace xgboost {
 
 const Version::TripletT Version::kInvalid {-1, -1, -1};
 
-Version::TripletT Version::Load(Json const& in, bool check) {
+Version::TripletT Version::Load(Json const& in) {
```
Comment (trivialfis) [inline src/common/version.cc:19]: What was I thinking when I added this parameter ...

File: src/c_api/c_api.cc [hunk 2]
```diff
@@ -163,7 +163,7 @@ XGB_DLL int XGDMatrixCreateFromCSCEx(const size_t* col_ptr,
                                      const unsigned* indices,
                                      const bst_float* data,
                                      size_t nindptr,
-                                     size_t nelem,
+                                     size_t,
```
Comment (trivialfis) [inline src/c_api/c_api.cc:166]: Please keep this parameter for now.  I'm not sure about these old APIs.

File: src/tree/tree_model.cc [hunk 3]
```diff
@@ -69,20 +69,24 @@ class TreeGenerator {
...
+                                int32_t /*nid*/, uint32_t /*depth*/) const {
     return "";
   }
-  virtual std::string Quantitive(RegTree const& tree, int32_t nid, uint32_t depth) const {
+  virtual std::string Quantitive(RegTree const& /*tree*/,
+                                int32_t /*nid*/, uint32_t /*depth*/) const {
     return "";
   }
-  virtual std::string NodeStat(RegTree const& tree, int32_t nid) const {
+  virtual std::string NodeStat(RegTree const& /*tree*/, int32_t /*nid*/) const {
```
Comment (trivialfis) [inline src/tree/tree_model.cc:84]: Thanks for converting them into comments!

File: src/c_api/c_api.cc (hunk 2 above)
Comment (igormp) [reply src/c_api/c_api.cc:166]: In a previous PR of mine I already left some parameters as unnamed in the C api, should I revert those as well?

Another option would be to use `__attribute__((unused))`, which should work on both GCC and Clang.

Comment (trivialfis) [reply src/c_api/c_api.cc:166]: Got it.  Then let's keep the PR as it's.  Thanks for the good work on reducing warnings!

**Review by trivialfis — APPROVED** [review]

File: src/tree/tree_model.cc (hunk 3 above)
Comment (igormp) [reply src/tree/tree_model.cc:84]: I thought that, since those are virtuals, it'd make sense to leave the name for anyone else doing some override :)

**Pull request merged**